### ISSUE TITLE
pAI Mega Issue Issue

### ISSUE THREAD

**Issue by Niknakflak** [tgstation/tgstation#48926]
Ninja, denton, and anne pressured me into this.  I am very sad this day came.  Made one issue because im lazy and horrible.  

1. pAIs can force themselves to be picked 
up in holoform.  Unsure if intentional or not but hilarious when i just jump into someones hands.  pAI drags themselves to another player

2. If a person puts a picked up pAI on their head as a hat and the pAI activates holomob form, the pAI will become a mob and also still be on the person's head.  The person can then take off the pAI sometimes and have duplicate pAIs until one disappears with some other fuckery involving turning into a holomob and back into card form.  Other times the pAI hat is stuck on them for awhile until they attempt to put a new hat on.

3. If a pAI is in something else, a locker, a mobs stomach, etc.  As card or holomob form, they can simply activate holomob form as a card to escape whatever containment they are in.  Ive done this inside a monkey stomach and then died an hour later when someone gibbed the monkey because apparently my mob was still tied to it.  Holomobs buggy. Hilarious, please do not fix. 

4. Host scan sometimes breaks if the above holomob stuff is done involving the hat or the escape method.  It might just break in general with holomob crap.

5. Using a meat hook on a pAI causes a weird interaction where it breaks the holomob but still hooks the mob.  I havent done this in awhile but if i remember correctly, it turns the pAI into the card.  Another player can then drag the hooked holomob that is separate from the actual pAI player and once the player turns into a holomob again, it will teleport the pAI player to wherever the holomob is.  I have teleported across the station and to different z levels after my mob was taken on the white shuttle.  Hilarious, please do not fix. 

6. pAIs are fire extinguishers.  If a pAI rubs against a mob on fire constantly, it will put the on fire mob out with no damage or change to the pAI.  pAIs can literally cuddle out someones fire.  Not sure if a bug and please do not fix this.

**Comment by trollbreeder** [comment]
Never before have i laughed as hard as i read all 6 of these issues. Do not fix any of them other than 2 and 4

**Comment by RaveRadbury** [comment]
Issue 1 is a reversal of what appears to be intended behavior. Players cannot initiate a pAI scooping by dragging the pAI onto them. I think the target and trigger got mixed up.

**Comment by zxaber** [comment]
Issue 2 is due to the buggy mob carrier object that is supposed to handle the interactions of being a hat item. Drones have the same issues (and are why drones and pAIs are blacklisted from being borg hats).

**Comment by kriskog** [comment]
1 seems intentional, and I have assumed as much. Being picked up as a holoform pAI even has a sprite, so its not entirely unintended.
2 and 3 happens from pretty much any contents I think, even if picked up in hand? I keep getting dupe pai cards whenever I have one at least.
4 is likely because of the dupe card, and its location being used.
5, well see above.

I will add you a 7: Empty pAI card ui windows do not have an adjacency check so you can keep the window open and accept a pAI to a card that is nowhere near you for the ultimate mindfuck. Might extend to doing stuff to the pAI after it is downloaded as well.

**Comment by Niknakflak** [comment]
1. Picking up is obviously intentional, its just whether it was intentional for the pAI to be one to initiate or was it supposed to be the human who initiates the pick up

**Comment by Archanial** [comment]
2 is duplicate of #47825

**Comment by Twaticus** [comment]
.1 is definitely backwards to how its supposed to work. it was added in #33442 and states that the user should be the one dragging the PAI onto themself.

not related to PAIs but its worth noting that fixing this bug would also fix the second half of #33442 which is picking ian up. right now the only way to get this to work is through an admin VV'ing hands onto a sentient ian so they can climb onto the person. the ian inhand sprites are too cute to never be achievable in game pls fix

**Comment by redguy999** [comment]
For 3 it makes sense that you would be killed if you aren’t suppose to get out that way.

Also how did you get in the stomach in the first place?

**Comment by TheBonded** [comment]
Moths can eat pAI holoforms

**Comment by Tharcoonvagh** [comment]
vorecode

**Comment by timothymtorres** [comment]
I tested some of these:

1.  Has been fixed.
2.  Has been fixed. 
3.  Partially fixed.  pAIs in lockers, mobs heads, etc. can still escape.  However the original location that they escaped from does not contain any copies of the pai anymore.
4.  I haven't tested but there is another issue report about it at #45662
5.  Has been fixed.  pAIs holoform cannot be hooked anymore.
6.  Is still a bug.  I set a mob on fire and as a pAI moved over the mob several times.  I noticed that each movement removed a fire stack from the person.  I suspect this is due to fire stacks transferring to another mob when they bump into each other.  Since pAIs can't catch fire, it effectively removes a fire stack each time.
